build/package/install_agent_spawn_host.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+set -ex
+
+while getopts 'i:h:t:u:v:' opt; do
+    case $opt in
+        i) keyfile="$OPTARG";;      # SSH identity file
+        u) user="$OPTARG";;         # Username on the remote host
+        h) hostsfile="$OPTARG";;    # Output of Evergreen host.list
+        t) groupid="$OPTARG";;      # APIGroupId
+        v) apiKey="$OPTARG";;       # APIKey
+        *) exit 1;;
+    esac
+done
+
+hosts=$(cat << EOF | python - "$hostsfile"
+import sys
+import json
+with open(sys.argv[1]) as hostsfile:
+    hosts = json.load(hostsfile)
+    for host in hosts:
+        print(host["dns_name"])
+EOF
+)
+for host in $hosts; do
+    set +e
+    echo "Installing dependeces on $host"
+    ssh -i "$keyfile" -o ConnectTimeout=50 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -tt "$user@$host" 'bash -s' <<'ENDSSH'
+        free -m
+        sudo apt-get install -y --no-install-recommends ca-certificates curl logrotate openssl snmp && exit
+ENDSSH
+    echo "Installing the automation agent on $host"
+    ssh -i "$keyfile" -o ConnectTimeout=50 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -tt "$user@$host" ARG1="$groupid" ARG2="$apiKey" 'bash -s' <<'ENDSSH'
```

Comment:
can we apply a similar thing to what ego does and centralise these args given you use them twice